kolibri/core/content/management/commands/deletecontent.py

```diff
@@ -170,6 +170,21 @@ def handle_async(self, *args, **options):
                 "Channel matching id {id} does not exist".format(id=channel_id)
             )
 
+        if node_ids and force_delete:
+            # As we are deleting similar nodes from everywhere
+            # We need to find the node_ids of all the similar nodes that are not passed from the frontend
+            # So we will find the content_ids of the nodes that are passed from the frontend and
+            # then find the node_ids of the nodes that have the same content_id(as content_id is common for all smilar nodes)
```

Comment:
I'm curious if maybe this addition isn't necessary - could you try testing it with these changes not applied?

I suspect that the validate method being fixed, which ensures that `force_delete` is passed to the task, then the feature should work as it did before. I mainly suspect this because it is a feature that used to work but no longer does.

I haven't looked at it closely, but I'd guess that `delete_metadata` uses the `force_delete` to the expected effect.

Reply:
That force_delete is behaving differently than my logic here. It deletes the entire channel when a single resource is force_deleted.
See In my `Imported Channel`. I have some content titled `Sample Video` that is present in both `Topic 1` and `Imported` Folder(They share the same content_id):
![Screenshot from 2024-09-26 11-28-45](https://github.com/user-attachments/assets/a60944fd-0382-4ad3-9229-c1789f5e0533)

So if we force_deleted the Sample Video, only that should have been deleted from both folders. But the entire channel gets deleted:
[Screencast from 26-09-24 11:31:00 AM IST.webm](https://github.com/user-attachments/assets/0c92e34f-22b9-4429-8789-33aec20309b5)


*Note*: This is only with the changes to DeleteChannelValidator *without* my change in the deletecontent command

Reply:
@thesujai thank you for that and apologies for my delay in responding here.

We've stumbled upon a chain of bugs... 😅 

This all started when I found that the checkbox to select "force_delete" had a bug, which then led you here to find the bug in the `DeleteChannelValidator#validate` method, which also led you to find that there is a bug with how we're handling granular deletion as well.

---

I was expecting the scope of this issue to be more along the lines of what the validator updates fix: _the user's selection is passed along to the queuing of the deletion task_. 

Let's get that fix merged in this PR and fixing the deletion issue itself can be done in a follow-up issue (which you're welcome to take on if you're interested).

Reply:
reverted! I can try fixing this behavior in a follow-up